service/hook/bitbucketv2/metrics.go

```diff
@@ -0,0 +1,169 @@
+package bitbucketv2
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/bitrise-io/bitrise-webhooks/service/hook/common"
+	"github.com/go-playground/webhooks/v6/bitbucket"
+)
+
+// GatherMetrics ...
+func (hp HookProvider) GatherMetrics(r *http.Request, appSlug string) ([]common.Metrics, error) {
+	hook, err := bitbucket.New()
+	if err != nil {
+		return nil, err
+	}
+
+	payload, err := hook.Parse(r, bitbucket.RepoPushEvent, bitbucket.PullRequestCreatedEvent, bitbucket.PullRequestUpdatedEvent, bitbucket.PullRequestMergedEvent, bitbucket.PullRequestDeclinedEvent)
+	if err != nil {
+		if err == bitbucket.ErrEventNotFound {
+			return nil, nil
+		}
+		return nil, err
+	}
+
+	event := r.Header.Get("X-Event-Key")
+	currentTime := hp.timeProvider.CurrentTime()
+
+	metricsList, err := hp.gatherMetrics(payload, event, appSlug, currentTime), nil
+	if err != nil {
+		return nil, err
+	}
+
+	return metricsList, nil
+}
+
+func (hp HookProvider) gatherMetrics(payload interface{}, webhookType, appSlug string, currentTime time.Time) []common.Metrics {
+	switch payload := payload.(type) {
```

Comment:
why not use `webhookType` instead of reflection?

Reply:
Even if the code would switch on `webhookType`, there is still a need for converting the `interface{}` returned by `hook.Parse` to a concrete type.
It seems the current type switching performs better than a reflection-based type conversion would: https://www.reddit.com/r/golang/comments/riki8m/type_switching_vs_simple_reflection_performance/

Also, this is the suggested way of working with this lib: https://github.com/go-playground/webhooks#examples .